derive/src/to_client.rs

```diff
@@ -0,0 +1,134 @@
+                    .iter().filter_map(|arg| {
+                        match arg {
+                            syn::FnArg::Captured(syn::ArgCaptured {
+                                pat: syn::Pat::Ident(syn::PatIdent { ident, .. }),
+                                ..
+                            }) => Some(ident),
+                            _ => None,
+                        }
+                    }).collect();
+                method_decl.push(syn::parse_quote! {
+                    fn #name(#args) -> #new_ty;
+                });
+                method_impl.push(syn::parse_quote! {
+                    fn #name(#args) -> #new_ty {
+                        let args_tuple = (#(#arg_names,)*);
+                        let args = serde_json::to_value(args_tuple)
+                            .expect("should never fail");
+                        let params = serde_json::from_value(args)
+                            .expect("should never fail");
+                        let request = Request::Single(Call::MethodCall(MethodCall {
+                            jsonrpc: Some(Version::V2),
+                            method: #rpc_name.to_owned(),
+                            params,
+                            id: Id::Num(next_id()),
+                        }));
+                        let request_str = serde_json::to_string(&request)
+                            .expect("should never fail");
+                        self.call_method(request_str)
+                            .and_then(rpc_impl_response::#name)
+                    }
+                });
+                method_resp.push(syn::parse_quote! {
+                    pub fn #name(response_str: String) -> FutureResult<#returns, Error> {
+                        let response = serde_json::from_str::<Response>(&response_str)
+                            .map_err(|_| Error::new(ErrorCode::ParseError));
+                        let response = match response {
+                            Ok(response) => response,
+                            Err(error) => return future::err(error),
+                        };
+                        let value: Result<Value> = match response {
+                            Response::Single(output) => output.into(),
+                            _ => return future::err(Error::parse_error()),
+                        };
+                        let value = match value {
+                            Ok(value) => value,
+                            Err(error) => return future::err(error),
+                        };
+                        let result = serde_json::from_value::<#returns>(value)
+                            .map_err(|_| Error::new(ErrorCode::ParseError));
+                        match result {
+                            Ok(result) => future::ok(result),
+                            Err(error) => future::err(error),
+                        }
+                    }
+                })
+            }
+            _ => {}
```

Comment:
I'd error out or at least print a warning in case of subscriptions if we generate the client, so that people know it's not supported.